File: languages/go/client/identity_core.go
package client

import (
	"github.com/bloock/bloock-sdk-go/v2/entity/identity"
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge"
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
	"github.com/bloock/bloock-sdk-go/v2/internal/config"
)

// IdentityCoreClient represents a client for interacting with the [Bloock Identity service].
//
// [Bloock Identity service]: https://bloock.com
type IdentityCoreClient struct {
	bridgeClient bridge.BloockBridge
	configData   *proto.ConfigData
}

// NewIdentityCoreClient creates a new instance of the IdentityCoreClient with default configuration.
func NewIdentityCoreClient() IdentityCoreClient {
	return IdentityCoreClient{
		bridgeClient: bridge.NewBloockBridge(),
		configData:   config.NewConfigDataDefault(),
	}
}

// NewIdentityCoreClientWithConfig creates a new instance of the IdentityCoreClient with the provided configuration.
func NewIdentityCoreClientWithConfig(configData *proto.ConfigData) IdentityCoreClient {
	return IdentityCoreClient{
		bridgeClient: bridge.NewBloockBridge(),
		configData:   configData,
	}
}

// BuildCredential creates a new credential builder for defining a credential on the Bloock Identity service.
func (c *IdentityCoreClient) BuildCredential(issuer identity.Issuer, schemaId, holderDid string, expiration int64, version int32) identity.CredentialCoreBuilder {
	return identity.NewCredentialCoreBuilder(issuer, schemaId, holderDid, expiration, version, c.configData)
}

File: languages/js/src/client/identity_core.ts
import { ConfigData } from "../bridge/proto/config";
import { NewConfigData } from "../config/config";
import { CredentialCoreBuilder, Issuer } from "../entity/identity";

/**
 * Represents a client for interacting with the [Bloock Identity service](https://dashboard.bloock.com/login).
 */
export class IdentityCoreClient {
    private configData: ConfigData;

    /**
     * Creates a new instance of the IdentityCoreClient with default configuration.
     * @param configData 
     */
    constructor(configData?: ConfigData) {
        this.configData = NewConfigData(configData);
    }

    /**
     * Creates a new credential builder for defining a credential on the Bloock Identity service.
     * @param issuer 
     * @param schemaId 
     * @param holderDid 
     * @param expiration 
     * @param version 
     * @returns 
     */
    public buildCredential(
        issuer: Issuer,
        schemaId: string,
        holderDid: string,
        expiration: number,
        version: number
    ): CredentialCoreBuilder {
        return new CredentialCoreBuilder(
            issuer,
            schemaId,
            holderDid,
            expiration,
            version,
            this.configData
        );
    }
}